crnk-core/src/main/java/io/crnk/core/engine/internal/jackson/JacksonPropertyNameResolver.java

```diff
@@ -0,0 +1,27 @@
+package io.crnk.core.engine.internal.jackson;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.crnk.core.engine.information.bean.BeanAttributeInformation;
+
+import java.util.Optional;
+
+/**
+ * Holds logic of retrieving property name based on {@link JsonProperty} value
+ */
+public class JacksonPropertyNameResolver {
```

Comment:
maybe let's also introduce a NameResolver interface (for that BeanInformationFactory brought up below). We could maybe slightly shortcut the name to JacksonNameResolver. It will allow to come up with other naming resolvers,  e.g. if somebody prefers x-y-z over camel case xYZ.